app/assets/javascripts/dashboard/views/dash/transfer_task_modal.js

```diff
@@ -0,0 +1,124 @@
+// @flow
+
+import _ from "lodash";
+import React from "react";
+import { Spin, Modal, Button } from "antd";
+import Request from "libs/request";
+import type { APIUserType } from "admin/api_flow_types";
+
+class TransferTaskModal extends React.PureComponent {
+  props: {
+    onChange: Function,
+    // Somehow, eslint doesn't recognize that annotationId is used in
+    // the async functions
+    // eslint-disable-next-line react/no-unused-prop-types
+    annotationId: ?string,
+    onCancel: Function,
+    visible: boolean,
+    userID: ?string,
+  };
+
+  state: {
+    isLoading: boolean,
+    users: Array<APIUserType>,
+    currentUserValue: string,
+  } = {
+    isLoading: false,
+    users: [],
+    currentUserValue: "",
+  };
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  async fetchData() {
+    this.setState({ isLoading: true });
+    const users = await Request.receiveJSON("/api/users");
+    this.setState({ isLoading: false });
+    const sortedUsers = _.sortBy(users, "lastName");
+
+    this.setState({
+      users: sortedUsers,
+    });
+  }
+
+  async transfer() {
+    if (!this.props.annotationId) {
+      throw new Error("No annotation id provided");
+    }
+    const url = `/annotations/Task/${this.props.annotationId}/transfer`;
+    this.setState({ isLoading: true });
+    await Request.sendJSONReceiveJSON(url, {
+      data: {
+        userId: this.state.currentUserValue,
+      },
+    });
+    this.setState({ isLoading: false });
+    this.props.onChange();
+  }
+
+  handleSelectChange = (event: SyntheticInputEvent) => {
+    this.setState({ currentUserValue: event.target.value });
+  };
+
+  renderFormContent() {
+    return (
+      <div>
+        <label htmlFor="transfer-user-select">{"New User's Name"}</label>
```

Comment:
Because there is a apostroph in the string, which does not work with plain text in jsx. Alternatively, we could write `New User{"'"} Name`, but I find this harder to read/write :)